client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -1529,35 +1529,22 @@ def _before_repre_load(self, placeholder, representation):
 
         pass
 
-    def _reduce_last_version_repre_entities(self, representations):
-        """Reduce representations to last verison."""
-
-        mapping = {}
-        # TODO use representation context with entities
-        # - using 'folder', 'subset' and 'version' from context on
-        #   representation is danger
-        for repre_entity in representations:
-            repre_context = repre_entity["context"]
-
-            folder_name = repre_context["asset"]
-            product_name = repre_context["subset"]
-            version = repre_context.get("version", -1)
-
-            if folder_name not in mapping:
-                mapping[folder_name] = {}
-
-            product_mapping = mapping[folder_name]
-            if product_name not in product_mapping:
-                product_mapping[product_name] = collections.defaultdict(list)
-
-            version_mapping = product_mapping[product_name]
-            version_mapping[version].append(repre_entity)
+    def _reduce_last_version_repre_entities(self, repre_contexts):
+        """Reduce representations to last version."""
+
+        version_mapping_by_product_id = {}
+        for repre_context in repre_contexts:
+            product_id = repre_context["product"]["id"]
+            version = repre_context["version"]["version"]
+            version_mapping = version_mapping_by_product_id.setdefault(
+                product_id, {}
+            )
+            version_mapping[version].append(repre_context)
 
         output = []
-        for product_mapping in mapping.values():
-            for version_mapping in product_mapping.values():
-                last_version = tuple(sorted(version_mapping.keys()))[-1]
-                output.extend(version_mapping[last_version])
+        for version_mapping in version_mapping_by_product_id.values():
+            last_version = tuple(sorted(version_mapping.keys()))[-1]
```

Comment:
Isn't this just the same as:
```suggestion
            last_version = max(version_mapping)
```